src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -147,22 +147,44 @@ class Simplex_tree {
     Filtration_value minval;
     Filtration_value maxval;
     Extended_filtration_data(){}
-    Extended_filtration_data(Filtration_value vmin, Filtration_value vmax): minval(vmin), maxval(vmax) {}
+    Extended_filtration_data(const Filtration_value& vmin, const Filtration_value& vmax): minval(vmin), maxval(vmax) {}
   };
   typedef typename std::conditional<Options::store_key, Key_simplex_base_real, Key_simplex_base_dummy>::type
       Key_simplex_base;
 
+  /**
+   * @brief Equals @ref Filtration_value if @ref SimplexTreeOptions::store_filtration is false or the type
+   * corresponds to a native arithmetic type (int, double, etc.). Equals to a reference of @ref Filtration_value
+   * otherwise.
```

Comment:
The idea was just to avoid referencing values which are faster to be copied than redirected. But I have no idea if that really brings that much performance wise. Removing this alias would make things easier, so if you don't think it is necessary, I will do so (in PR #1122 first, I will not really work on here before finishing the others).

Reply:
Most small functions are inlined, and after inlining it doesn't matter if the int was passed by reference or value. For most big functions, the time taken by an indirection for an int parameter is negligible. So trying to pass by value conditionally is not worth it. If the type is explicitly `int`, sure, pass by value (less typing). But if it is a template that may sometimes be heavy and require a reference, let's *always* use the reference.